### PR TITLE
Todo React app

### PR DESCRIPTION
**Pull request by devMahmoud**
## In this branch I did:
* Add components for the todos list, form, and todo item
* Add styling to the components
* Add functionalities for adding, updating, and deleting a todo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
+"React Todo App" is a simple React application for managing todos. It is a Single Page App (SPA) that allows users to:
 
-* Make simple calculations.
-* Read a random math-related quote.
+* Add todo.
+* Edit todo.
+* Remove todo.
 
 
 ## Built With
```
Comment (rskinnerc) [inline README.md:11]: Please note the Live Demo link is not working.

File: src/components/Header.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/* eslint-disable */
```
Comment (rskinnerc) [inline src/components/Header.js]: Please don't disable linters for a whole file. The purpose of linters is to warn about not following a coding pattern.

File: src/components/InputTodo.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable */
+import React, { Component } from 'react';
+
+class InputTodo extends Component {
```
Comment (rskinnerc) [inline src/components/InputTodo.js:4]: Please consider implementing the function component style and hooks

File: src/components/TodoItem.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
+/* eslint-disable */
```
Comment (rskinnerc) [inline src/components/TodoItem.js]: Please don't disable ES lint

File: src/components/TodoContainer.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
+/* eslint-disable */
```
Comment (rskinnerc) [inline src/components/TodoContainer.js]: Please don't disable ES lint

File: src/components/TodosList.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable */
```
Comment (rskinnerc) [inline src/components/TodosList.js]: Please don't disable ES lint

File: src/index.js [hunk 7]
```diff
@@ -1,17 +1,16 @@
...
-import App from './App';
+import TodoContainer from './components/TodoContainer';
 import reportWebVitals from './reportWebVitals';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
   <React.StrictMode>
-    <App />
-  </React.StrictMode>
+    <TodoContainer />
```
Comment (rskinnerc) [inline src/index.js:10]: Kindly consider implementing the router part of the tutorial since it is not optional

**Review by rskinnerc — CHANGES_REQUESTED** [review]
## Changes required :recycle: 

Hi, @devMahmoud 
Good job! 

quality-quality-work

Your code is clean and everything is working fine but there are small changes that could make it even better

## (Highlights) Good Points: :+1: 
* You documented your work professionally. :heavy_check_mark:
* Gitflow is used correctly. :heavy_check_mark:

## [Optional] suggestions:

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding! :clap::clap::clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

**Comment by devMahmoud** [comment]
Thanks for the review :+1:

File: src/components/TodoContainer.js [hunk 8]
```diff
@@ -1,77 +1,79 @@
-/* eslint-disable */
+/* eslint-disable no-param-reassign */
+/* eslint-disable react/no-access-state-in-setstate */
+/* eslint-disable react/destructuring-assignment */
```
Comment (rskinnerc) [inline src/components/TodoContainer.js]: Dear @devMahmoud disabling all of them is like disabling the whole file. Try to correct the ESLint errors in following projects

**Review by rskinnerc — CHANGES_REQUESTED** [review]
@devMahmoud 

There are still some required changes.

Please do not disable all linter errors. It's pretty much the same than disabling for the whole file. 

Kindly try to correct all of them and proceed with the merge of your code.

Please do not hesitate on contacting me for any doubt.

**Review by rskinnerc — APPROVED** [review]
Excellent job @devMahmoud 

Everything seems very well after the changes. Keep your hard work up!

**Pull request merged**